### PR TITLE
🛑 [never merge] plan document for /css folder

### PR DESCRIPTION
**Pull request by wibjorn**
# Welcome to the plan for the /css folder

The main point of this pull request is to provide a place for use look at the plan document together. This PR will never merge. The plan document is here: 

https://github.com/microsoft/atlas-design/pull/2/files#diff-7b7df4436312ddd4bf006ef1f51da2644ab69a5a58406df81a77d7321bb21e33

task-376343

Note: there are several changes in the files changed tab. These can be ignored, I've been using the PR to proof of concept certain ideas well, including changesets.

### CODE REVIEW

**Comment by wibjorn** [comment]
ticket-376343

File: css-technical-plan.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+- theming
+- no classes, except theme
+
+#### Atomics
+
+- single classes that have a very targeted effect
+- generally 1 effect - 1 css property
+  - with the exception outline color and backgrounds (for focus)
+- naming should match property names and values as much as possible
+- standardize on `has` or impose new prefix
```
Comment (squalrus) [inline css-technical-plan.md:126]: 👍

File: css-technical-plan.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+#### Mixins
+
+- No output
+- Depends on tokens
+- Contains shared mixins and functions
+
+#### Core
+
+- output producing
```
Comment (squalrus) [inline css-technical-plan.md:113]: Is it worth noting "output producing" for "Atomics" and "Components", or is it implied/intuitive?

File: css-technical-plan.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  - 👨‍💻 create css custom properties for theme colors
+  - 👨‍💻 implement font stack
+    - no font face rules
+    - just native font family (from github / bulma)
+    - this will be overridden downstream
+  - 👨‍💻 implement focus styling (focus-visible) 💭
+    - escape hatch, which can toggle :focus-visible, rather than the polyfill (.focus-visible)
+  - 👨‍💻 implement a css build for prototyping 💭
+    - should include a font face?
+- devops setup
```
Comment (squalrus) [inline css-technical-plan.md:54]: Is the site in scope for this, or is it strictly the CSS for Atlas?

**Review by squalrus — COMMENTED** [review]
Awesome! Thank you for putting this together!

File: css-technical-plan.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  - add top level package.json and init monorepo
+- 👨‍💻 implment changesets [changesets](https://github.com/atlassian/changesets)
+- 👨‍💻 implement tokens (each direct sub list item is a dev task)
+  - color tokens
+    - convert tokens to hsla (pending dropping IE theming confirmation)
+    - raw colors in palette.scss
+    - themed colors in color-themes.scss
+  - 👨‍💻 spacing tokens
+  - 👨‍💻 typography, font-size tokens
+  - 👨‍💻 depth/shadow tokens
```
Comment (5t3ph) [inline css-technical-plan.md:31]: For future consideration: z-index tokens

File: css-technical-plan.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  - 👨‍💻 typography atomic classes
+  - 👨‍💻 color atomic classes
+  - 💭 standardize on has
+- create components folder
+  - create components folder readme and index file
+  - Will we be implementing any components as part of this feature 💭
+    - 👨‍💻 columns
+    - 👨‍💻 nav-bar?
+- 👨‍💻 plugins (top level folder)
+  - this folder contains postcss plugins
```
Comment (5t3ph) [inline css-technical-plan.md:71]: Is this intended to mean custom plugins? will it also hold configs for plugins in use like autoprefixer?

File: css-technical-plan.md (hunk 3 above)
Comment (wibjorn) [reply css-technical-plan.md:54]: Not for this plan, this is just the init of the css

File: css-technical-plan.md (hunk 2 above)
Comment (wibjorn) [reply css-technical-plan.md:113]: I think it's worth noting, I'll do that

File: css-technical-plan.md (hunk 4 above)
Comment (wibjorn) [reply css-technical-plan.md:31]: We have some already in docs-ui, I see no reason why they can't move initially.

Comment (wibjorn) [reply css-technical-plan.md:31]: They look like this:

```
// State related z-index
$zindex-active: 1;
$zindex-hover: 2;
$zindex-focus: 3;
$zindex-multi: 4;
// Component type related z-index
$zindex-dropdown: 1000;
$zindex-sticky: 1020;
$zindex-fixed: 1030;
$zindex-modal-backdrop: 1040;
$zindex-modal: 1050;
$zindex-popover: 1060;
$zindex-tooltip: 1070;

File: css-technical-plan.md (hunk 5 above)
Comment (wibjorn) [reply css-technical-plan.md:71]: yes, custom plugins only I would think, in particular we have a IE compat related one that would be the only one to go in initially

File: package.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+		"design",
+		"css",
+		"monorepo",
+		"microsoft"
+	],
+	"workspaces": [
+		"./css/"
+	],
+	"author": "",
+	"license": "MIT",
```
[the remark this answers is not in the document]
Comment (wibjorn) [reply package.json:24]: What's involved is determining the level of protection/restriction of reuse of the framework. MIT is a pretty standard permissive license, and the default selection for projects generated by the Microsoft open source tool

**Pull request closed without merging**